docs/learn/voting-dapp.mdx

```diff
@@ -575,33 +575,82 @@ if (!candidates) return;
 
 ```jsx
 async function vote() {
-  if (!walletClient || !address || number === "") {
-    console.error("Missing required data for voting");
-    return;
-  }
-  if (
-    isNaN(Number(number)) ||
-    Number(number) < 0 ||
-    Number(number) >= candidates.length
-  ) {
-    console.error("Invalid candidate index");
-    return;
-  }
-  if (!votingStatus) {
-    console.error("Voting is not currently active");
-    return;
-  }
   try {
-    console.log("Attempting to vote for candidate index:", number);
-    writeContract({
+    // Basic validation checks
+    if (!walletClient) {
```

Comment:
Thank you for the PR :)

Mind if I ask the rationale for placing input validation inside the try... clause?

From my POV, the code is clearer with input validation before the try...catch... clause